apps/emqx_bridge_pgsql/src/emqx_bridge_pgsql.erl

```diff
@@ -1,83 +1,90 @@
 -include_lib("hocon/include/hoconsc.hrl").
+-include_lib("epgsql/include/epgsql.hrl").
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
 -include_lib("emqx_resource/include/emqx_resource.hrl").
 
--import(hoconsc, [mk/2, enum/1, ref/2]).
-
--export([
-    conn_bridge_examples/1,
-    values/2,
-    fields/2
-]).
-
 -export([
     namespace/0,
     roots/0,
     fields/1,
-    desc/1
+    desc/1,
+    fields/2
 ]).
 
--define(DEFAULT_SQL, <<
-    "insert into t_mqtt_msg(msgid, topic, qos, payload, arrived) "
-    "values (${id}, ${topic}, ${qos}, ${payload}, TO_TIMESTAMP((${timestamp} :: bigint)/1000))"
->>).
+%% Examples
+-export([
+    bridge_v2_examples/1,
+    conn_bridge_examples/1
+]).
 
-%% -------------------------------------------------------------------------------------------------
-%% api
+%% Exported for timescale and matrix bridges
+-export([
+    values/1,
+    values_conn_bridge_examples/2
+]).
 
-conn_bridge_examples(Method) ->
-    [
-        #{
-            <<"pgsql">> => #{
-                summary => <<"PostgreSQL Bridge">>,
-                value => values(Method, pgsql)
-            }
-        }
-    ].
+-define(PGSQL_HOST_OPTIONS, #{
+    default_port => ?PGSQL_DEFAULT_PORT
+}).
 
-values(_Method, Type) ->
-    #{
-        enable => true,
-        type => Type,
-        name => <<"foo">>,
-        server => <<"127.0.0.1:5432">>,
-        database => <<"mqtt">>,
-        pool_size => 8,
-        username => <<"root">>,
-        password => <<"******">>,
-        sql => ?DEFAULT_SQL,
-        local_topic => <<"local/topic/#">>,
-        resource_opts => #{
-            worker_pool_size => 8,
-            health_check_interval => ?HEALTHCHECK_INTERVAL_RAW,
-            batch_size => ?DEFAULT_BATCH_SIZE,
-            batch_time => ?DEFAULT_BATCH_TIME,
-            query_mode => async,
-            max_buffer_bytes => ?DEFAULT_BUFFER_BYTES
-        }
-    }.
-
-%% -------------------------------------------------------------------------------------------------
 %% Hocon Schema Definitions
 namespace() -> "bridge_pgsql".
 
-roots() -> [].
-
+roots() ->
+    [].
+
+fields("config_connector") ->
+    emqx_postgresql_connector_schema:fields("config_connector");
+fields(config) ->
+    fields("config_connector") ++
+        fields(action);
+fields(action) ->
+    {pgsql,
+        hoconsc:mk(
+            hoconsc:map(name, hoconsc:ref(emqx_bridge_pgsql, pgsql_action)),
+            #{
+                desc => <<"PostgreSQL Action Config">>,
+                required => false
+            }
+        )};
+fields(action_parameters) ->
+    [
+        {sql,
+            hoconsc:mk(
+                binary(),
+                #{desc => ?DESC("sql_template"), default => default_sql(), format => <<"sql">>}
+            )}
+    ] ++
+        emqx_connector_schema_lib:prepare_statement_fields();
+fields(pgsql_action) ->
+    emqx_bridge_v2_schema:make_producer_action_schema(hoconsc:ref(?MODULE, action_parameters));
```

Comment:
I thought this should be:

```suggestion
    emqx_bridge_v2_schema:make_producer_action_schema(hoconsc:mk(hoconsc:ref(?MODULE, action_parameters), #{required => true, desc => ?DESC(parameters)));
```

to be a valid hocon schema type?  :thinking:

Reply:
I think it works because make_producer_action_schema wraps things in the right way

Reply:
From what I can see in the current version of `make_producer_action_schema`, I think it's just fortuitous it's working now.  But if we look at hocon types, it _should_ be wrapped in `hoconsc:mk`, I believe.

```erlang
-callback fields(name()) -> fields().

-type field() :: {name(), typefunc() | field_schema()}.
-type fields() ::
    [field()]
    | #{
        fields := [field()],
        desc => desc()
    }.

-type field_schema() ::
    typerefl:type()
    | ?UNION(union_members())
    | ?ARRAY(type())
    | ?ENUM(type())
    | field_schema_map()
    | field_schema_fun().
```
(and `hoconsc:ref/2` returns just a `{ref, _, _}` tuple)

Reply:
Ok, I will try to wrap it. If it still works then all is good.